pkg/reader/ccip.go

```diff
@@ -550,6 +551,63 @@ func (r *ccipChainReader) GetChainFeePriceUpdate(ctx context.Context, selectors
 	return feeUpdates
 }
 
+func (r *ccipChainReader) GetRMNRemoteConfig(
+	ctx context.Context,
+	destChainSelector cciptypes.ChainSelector,
+) (rmntypes.RemoteConfig, error) {
+	if err := validateExtendedReaderExistence(r.contractReaders, destChainSelector); err != nil {
+		return rmntypes.RemoteConfig{}, err
+	}
+
+	contractAddress, err := r.GetContractAddress(consts.ContractNameRMNRemote, destChainSelector)
+	if err != nil {
+		return rmntypes.RemoteConfig{}, fmt.Errorf("get RMNRemote contract address: %w", err)
+	}
+
+	var vc versionnedConfig
+	err = r.contractReaders[destChainSelector].ExtendedGetLatestValue(
+		ctx,
+		consts.ContractNameRMNRemote,
+		consts.MethodNameGetVersionedConfig,
+		primitives.Unconfirmed,
+		map[string]any{},
+		&vc,
+	)
+	if err != nil {
+		return rmntypes.RemoteConfig{}, fmt.Errorf("get RMNRemote config: %w", err)
+	}
+
+	var dh string
+	err = r.contractReaders[destChainSelector].ExtendedGetLatestValue(
+		ctx,
+		consts.ContractNameRMNRemote,
+		consts.MethodNameGetReportDigestHeader,
+		primitives.Unconfirmed,
+		map[string]any{},
+		&dh,
+	)
+	if err != nil {
+		return rmntypes.RemoteConfig{}, fmt.Errorf("get RMNRemote report digest header: %w", err)
+	}
+
+	signers := make([]rmntypes.RemoteSignerInfo, 0, len(vc.Config.Signers))
+	for _, signer := range vc.Config.Signers {
+		signers = append(signers, rmntypes.RemoteSignerInfo{
+			OnchainPublicKey: signer.OnchainPublicKey,
+			NodeIndex:        signer.NodeIndex,
+		})
+	}
+
+	return rmntypes.RemoteConfig{
+		ContractAddress:  contractAddress,
+		ConfigDigest:     cciptypes.Bytes32(vc.Config.RMNHomeContractConfigDigest),
+		Signers:          signers,
+		MinSigners:       vc.Config.MinSigners,
+		ConfigVersion:    vc.Version,
+		RmnReportVersion: dh,
```

Comment:
I think there's a mismatch b/w what you're expecting and what the contract returns here.

```solidity
bytes32 private constant RMN_V1_6_ANY2EVM_REPORT = keccak256("RMN_V1_6_ANY2EVM_REPORT");

function getReportDigestHeader() external pure returns (bytes32 digestHeader) {
  return RMN_V1_6_ANY2EVM_REPORT;
}
```

`RMN_V1_6_ANY2EVM_REPORT` is a hash - so this field should be `cciptypes.Bytes32` ideally.

Reply:
Good catch